src/lib.rs

```diff
@@ -499,14 +512,506 @@ impl RadiumBool for Cell<bool> {
+        failure: Ordering,
+    ) -> Result<Item, Item>;
+
+    /// Stores a value into this object if the currently-stored value is the
+    /// same as the `current` value.
+    ///
+    /// Unlike `compare_exchange`, this function is allowed to spuriously fail
+    /// even when the comparison succeeds, which can result in more efficient
+    /// code on some platforms. The return value is a `Result` indicating
+    /// whether the new value was written, and containing the previously-stored
+    /// value.
+    ///
+    /// Ordering arguments are ignored by non-atomic types.
+    ///
+    /// See also: [`AtomicU8::compare_exchange_weak`].
+    ///
+    /// [`AtomicU8::compare_exchange_weak`]: core::sync::atomic::AtomicU8::compare_exchange_weak
+    fn compare_exchange_weak(
+        &self,
+        current: Item,
+        new: Item,
+        success: Ordering,
+        failure: Ordering,
+    ) -> Result<Item, Item>;
+}
+
+/// Performs bitwise arithmetic on a maybe-atomic value.
+pub trait RadiumBits<Item>: Radium<Item> {
+    /// Performs a bitwise "and" on the currently-stored value and the argument
+    /// `value`, and stores the result in `self`.
+    ///
+    /// Returns the previously-stored value.
+    ///
+    /// Ordering arguments are ignored by non-atomic types.
+    ///
+    /// See also: [`AtomicU8::fetch_and`].
+    ///
+    /// [`AtomicU8::fetch_and`]: core::sync::atomic::AtomicU8::fetch_and
+    fn fetch_and(&self, value: Item, order: Ordering) -> Item;
+
+    /// Performs a bitwise "nand" on the currently-stored value and the argument
+    /// `value`, and stores the result in `self`.
+    ///
+    /// Returns the previously-stored value.
+    ///
+    /// Ordering arguments are ignored by non-atomic types.
+    ///
+    /// See also: [`AtomicU8::fetch_nand`].
+    ///
+    /// [`AtomicU8::fetch_nand`]: core::sync::atomic::AtomicU8::fetch_nand
+    fn fetch_nand(&self, value: Item, order: Ordering) -> Item;
+
+    /// Performs a bitwise "or" on the currently-stored value and the argument
+    /// `value`, and stores the result in `self`.
+    ///
+    /// Returns the previously-stored value.
+    ///
+    /// Ordering arguments are ignored by non-atomic types.
+    ///
+    /// See also: [`AtomicU8::fetch_or`].
+    ///
+    /// [`AtomicU8::fetch_or`]: core::sync::atomic::AtomicU8::fetch_or
+    fn fetch_or(&self, value: Item, order: Ordering) -> Item;
+
+    /// Performs a bitwise "xor" on the currently-stored value and the argument
+    /// `value`, and stores the result in `self`.
+    ///
+    /// Returns the previously-stored value.
+    ///
+    /// Ordering arguments are ignored by non-atomic types.
+    ///
+    /// See also: [`AtomicU8::fetch_xor`].
+    ///
+    /// [`AtomicU8::fetch_xor`]: core::sync::atomic::AtomicU8::fetch_xor
+    fn fetch_xor(&self, value: Item, order: Ordering) -> Item;
+}
+
+/// Performs integer arithmetic on a maybe-atomic value.
+pub trait RadiumInteger<Item>: Radium<Item> {
```

Comment:
`RadiumInteger` implies `RadiumBits`, so we can make them inherit :-)

```suggestion
pub trait RadiumInteger<Item>: RadiumBits<Item> {
```